figures/commits_figure.py

```diff
@@ -0,0 +1,116 @@
+"""
+This is a subset of the functionality in astropy/astropy-tools in the
+visualizations_demographics subdirectory.
+"""
+
+from datetime import datetime
+import os
+import pandas as pd
+import numpy as np
+from matplotlib import pyplot as plt
+
+def generate_commit_stats_file(filename='gitlogstats.csv', overwrite=False,
+                               astropy_path=None):
+    """
+    Running this will generate a file in the current directory that stores the
+    statistics to make generating lots of plots easier.  Delete the file or
+    set `overwrite` to True to always re-generate the statistics.
+    """
+    if os.path.isfile(filename) and not overwrite:
+        return pd.read_csv(filename)
+
+    else:
+        import subprocess
+        import tempfile
+
+        # cmd = 'git log --use-mailmap --no-merges --format=format:\"%H\",\"%aI\",\"%aN\",\"%cN\"'.split()
+        cmd = 'git log --use-mailmap --format=format:\"%H\",\"%aI\",\"%aN\",\"%cN\"'.split()
```

Comment:
Just as an FYI, you could do ``git shortlog --summary --numbered`` - that does the job too and is easier to remember... (that's what I did)